@Test public void testDoOnError(){
  final AtomicReference<Throwable> r=new AtomicReference<Throwable>();
  Throwable t=null;
  try {
    Observable.<String>error(new RuntimeException("an error")).doOnError(new Consumer<Throwable>(){
      @Override public void accept(      Throwable v){
        r.set(v);
      }
    }
).blockingSingle();
    fail("expected exception, not a return value");
  }
 catch (  Throwable e) {
    t=e;
  }
  assertNotNull(t);
  assertEquals(t,r.get());
}
